lib/widgets/components/gravatar_image.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:coffeecard/utils/encode_passcode.dart';
+import 'package:coffeecard/widgets/components/helpers/shimmer_builder.dart';
+import 'package:flutter/material.dart';
+
+class GravatarImage extends StatelessWidget {
+  final String hash;
+  final double size;
+
+  const GravatarImage.large({required this.hash}) : size = 100;
+
+  const GravatarImage.small({required this.hash}) : size = 40;
+
+  @override
+  Widget build(BuildContext context) {
+    return CachedNetworkImage(
+      imageBuilder: (context, imageProvider) => Container(
+        width: size,
+        height: size,
+        decoration: BoxDecoration(
+          shape: BoxShape.circle,
+          image: DecorationImage(image: imageProvider, fit: BoxFit.cover),
+        ),
+      ),
+      imageUrl:
+          'https://gravatar.com/avatar/${hashMD5(hash)}?s=100&d=identicon',
```

Comment:
This might be an over-engineering suggestion, but to plan a bit for the future, then would it make sense to wrap this call in a Repository class as we do for other RestApi calls such that the UI do not care about this task

Reply:
Removed Gravatar in favor of colored background